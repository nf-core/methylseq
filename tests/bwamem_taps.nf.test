nextflow_pipeline {

    name "Test Workflow main.nf - BWA-MEM with TAPS"
    script "../main.nf"
    config "./nextflow.config"
    tag "cpu"

    test("Params: bwamem | taps") {
        when {
            params {
                aligner = "bwamem"
                taps    = true
                outdir  = "$outputDir"
            }
        }

        then {
            // stable_name: All files + folders in ${params.outdir}/ with a stable name
            def stable_name = getAllFilesFromDir(params.outdir, relative: true, includeDir: true, ignore: ['pipeline_info/*.{html,json,txt}'])
            // stable_path: All files in ${params.outdir}/ with stable content
            def stable_path = getAllFilesFromDir(params.outdir, ignoreFile: 'tests/.nftignore')
            // bam_files: All bam files
            def bam_files  = getAllFilesFromDir(params.outdir, include: ['**/*.bam'])
            assertAll(
                { assert workflow.success},
                { assert snapshot(
                    // Number of tasks
                    workflow.trace.succeeded().size(),
                    // pipeline versions.yml file for multiqc from which Nextflow version is removed because we tests pipelines on multiple Nextflow versions
                    removeNextflowVersion("$outputDir/pipeline_info/nf_core_methylseq_software_mqc_versions.yml"),
                    // All stable path name
                    stable_name,
                    // All files with stable contents
                    stable_path,
                    // All bam files
                    bam_files.collect{ file -> [ file.getName(), bam(file.toString()).getReadsMD5() ] }
                ).match() }
            )
        }
    }

}
