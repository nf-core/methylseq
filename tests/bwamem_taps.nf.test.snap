{
    "Params: bwamem | taps": {
        "content": [
            51,
            {
                "BWA_INDEX": {
                    "bwa": "0.7.18-r1243-dirty"
                },
                "BWA_MEM": {
                    "bwa": "0.7.18-r1243-dirty",
                    "samtools": 1.21
                },
                "CONVERT_TO_METHYLKIT": {
                    "rastair": "rastair 0.8.2"
                },
                "FASTQC": {
                    "fastqc": "0.12.1"
                },
                "GUNZIP": {
                    "gunzip": 1.13
                },
                "RASTAIR_CALL": {
                    "rastair": "rastair 0.8.2"
                },
                "RASTAIR_MBIAS": {
                    "rastair": "rastair 0.8.2"
                },
                "RASTAIR_MBIAS_PARSER": {
                    "rastair": "rastair 0.8.2"
                },
                "TRIMGALORE": {
                    "trimgalore": "0.6.10",
                    "cutadapt": 4.9,
                    "pigz": 2.8
                },
                "Workflow": {
                    "nf-core/methylseq": "v4.1.0"
                }
            },
            [
                "bwa",
                "bwa/Ecoli_10K_methylated.bam",
                "bwa/SRR389222_sub1.bam",
                "bwa/SRR389222_sub2.bam",
                "bwa/SRR389222_sub3.bam",
                "bwa/bwa",
                "bwa/bwa/genome.amb",
                "bwa/bwa/genome.ann",
                "bwa/bwa/genome.bwt",
                "bwa/bwa/genome.pac",
                "bwa/bwa/genome.sa",
                "bwamem",
                "bwamem/alignments",
                "bwamem/alignments/samtools_stats",
                "bwamem/alignments/samtools_stats/Ecoli_10K_methylated.flagstat",
                "bwamem/alignments/samtools_stats/Ecoli_10K_methylated.stats",
                "bwamem/alignments/samtools_stats/SRR389222_sub1.flagstat",
                "bwamem/alignments/samtools_stats/SRR389222_sub1.stats",
                "bwamem/alignments/samtools_stats/SRR389222_sub2.flagstat",
                "bwamem/alignments/samtools_stats/SRR389222_sub2.stats",
                "bwamem/alignments/samtools_stats/SRR389222_sub3.flagstat",
                "bwamem/alignments/samtools_stats/SRR389222_sub3.stats",
                "bwamem/deduplicated",
                "bwamem/deduplicated/Ecoli_10K_methylated.dedup.sorted.bam",
                "bwamem/deduplicated/Ecoli_10K_methylated.dedup.sorted.bam.bai",
                "bwamem/deduplicated/Ecoli_10K_methylated.deduplicated.sorted.bam",
                "bwamem/deduplicated/Ecoli_10K_methylated.deduplicated.sorted.bam.bai",
                "bwamem/deduplicated/SRR389222_sub1.dedup.sorted.bam",
                "bwamem/deduplicated/SRR389222_sub1.dedup.sorted.bam.bai",
                "bwamem/deduplicated/SRR389222_sub1.deduplicated.sorted.bam",
                "bwamem/deduplicated/SRR389222_sub1.deduplicated.sorted.bam.bai",
                "bwamem/deduplicated/SRR389222_sub2.dedup.sorted.bam",
                "bwamem/deduplicated/SRR389222_sub2.dedup.sorted.bam.bai",
                "bwamem/deduplicated/SRR389222_sub2.deduplicated.sorted.bam",
                "bwamem/deduplicated/SRR389222_sub2.deduplicated.sorted.bam.bai",
                "bwamem/deduplicated/SRR389222_sub3.dedup.sorted.bam",
                "bwamem/deduplicated/SRR389222_sub3.dedup.sorted.bam.bai",
                "bwamem/deduplicated/SRR389222_sub3.deduplicated.sorted.bam",
                "bwamem/deduplicated/SRR389222_sub3.deduplicated.sorted.bam.bai",
                "bwamem/deduplicated/picard_metrics",
                "bwamem/deduplicated/picard_metrics/Ecoli_10K_methylated.dedup.sorted.MarkDuplicates.metrics.txt",
                "bwamem/deduplicated/picard_metrics/SRR389222_sub1.dedup.sorted.MarkDuplicates.metrics.txt",
                "bwamem/deduplicated/picard_metrics/SRR389222_sub2.dedup.sorted.MarkDuplicates.metrics.txt",
                "bwamem/deduplicated/picard_metrics/SRR389222_sub3.dedup.sorted.MarkDuplicates.metrics.txt",
                "fastqc",
                "fastqc/Ecoli_10K_methylated_1_fastqc.html",
                "fastqc/Ecoli_10K_methylated_2_fastqc.html",
                "fastqc/SRR389222_sub1_fastqc.html",
                "fastqc/SRR389222_sub2_fastqc.html",
                "fastqc/SRR389222_sub3_fastqc.html",
                "fastqc/zips",
                "fastqc/zips/Ecoli_10K_methylated_1_fastqc.zip",
                "fastqc/zips/Ecoli_10K_methylated_2_fastqc.zip",
                "fastqc/zips/SRR389222_sub1_fastqc.zip",
                "fastqc/zips/SRR389222_sub2_fastqc.zip",
                "fastqc/zips/SRR389222_sub3_fastqc.zip",
                "multiqc",
                "multiqc/bwamem",
                "multiqc/bwamem/multiqc_data",
                "multiqc/bwamem/multiqc_data/BETA-multiqc.parquet",
                "multiqc/bwamem/multiqc_data/cutadapt_filtered_reads_plot.txt",
                "multiqc/bwamem/multiqc_data/cutadapt_trimmed_sequences_plot_3_Counts.txt",
                "multiqc/bwamem/multiqc_data/cutadapt_trimmed_sequences_plot_3_Obs_Exp.txt",
                "multiqc/bwamem/multiqc_data/fastqc-status-check-heatmap.txt",
                "multiqc/bwamem/multiqc_data/fastqc_adapter_content_plot.txt",
                "multiqc/bwamem/multiqc_data/fastqc_overrepresented_sequences_plot.txt",
                "multiqc/bwamem/multiqc_data/fastqc_per_base_n_content_plot.txt",
                "multiqc/bwamem/multiqc_data/fastqc_per_base_sequence_quality_plot.txt",
                "multiqc/bwamem/multiqc_data/fastqc_per_sequence_gc_content_plot_Counts.txt",
                "multiqc/bwamem/multiqc_data/fastqc_per_sequence_gc_content_plot_Percentages.txt",
                "multiqc/bwamem/multiqc_data/fastqc_per_sequence_quality_scores_plot.txt",
                "multiqc/bwamem/multiqc_data/fastqc_sequence_counts_plot.txt",
                "multiqc/bwamem/multiqc_data/fastqc_sequence_duplication_levels_plot.txt",
                "multiqc/bwamem/multiqc_data/fastqc_top_overrepresented_sequences_table.txt",
                "multiqc/bwamem/multiqc_data/llms-full.txt",
                "multiqc/bwamem/multiqc_data/multiqc.log",
                "multiqc/bwamem/multiqc_data/multiqc_citations.txt",
                "multiqc/bwamem/multiqc_data/multiqc_cutadapt.txt",
                "multiqc/bwamem/multiqc_data/multiqc_data.json",
                "multiqc/bwamem/multiqc_data/multiqc_fastqc.txt",
                "multiqc/bwamem/multiqc_data/multiqc_general_stats.txt",
                "multiqc/bwamem/multiqc_data/multiqc_picard_dups.txt",
                "multiqc/bwamem/multiqc_data/multiqc_samtools_flagstat.txt",
                "multiqc/bwamem/multiqc_data/multiqc_samtools_idxstats.txt",
                "multiqc/bwamem/multiqc_data/multiqc_samtools_stats.txt",
                "multiqc/bwamem/multiqc_data/multiqc_software_versions.txt",
                "multiqc/bwamem/multiqc_data/multiqc_sources.txt",
                "multiqc/bwamem/multiqc_data/picard_MarkIlluminaAdapters_histogram.txt",
                "multiqc/bwamem/multiqc_data/picard_MeanQualityByCycle_histogram.txt",
                "multiqc/bwamem/multiqc_data/picard_QualityScoreDistribution_histogram.txt",
                "multiqc/bwamem/multiqc_data/picard_deduplication.txt",
                "multiqc/bwamem/multiqc_data/samtools-flagstat-pct-table.txt",
                "multiqc/bwamem/multiqc_data/samtools-flagstat-table.txt",
                "multiqc/bwamem/multiqc_data/samtools-idxstats-mapped-reads-plot_Normalised_Counts.txt",
                "multiqc/bwamem/multiqc_data/samtools-idxstats-mapped-reads-plot_Observed_over_Expected_Counts.txt",
                "multiqc/bwamem/multiqc_data/samtools-idxstats-mapped-reads-plot_Raw_Counts.txt",
                "multiqc/bwamem/multiqc_data/samtools-stats-dp.txt",
                "multiqc/bwamem/multiqc_data/samtools_alignment_plot.txt",
                "multiqc/bwamem/multiqc_plots",
                "multiqc/bwamem/multiqc_plots/pdf",
                "multiqc/bwamem/multiqc_plots/pdf/cutadapt_filtered_reads_plot-cnt.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/cutadapt_filtered_reads_plot-pct.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/cutadapt_trimmed_sequences_plot_3_Counts.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/cutadapt_trimmed_sequences_plot_3_Obs_Exp.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/fastqc-status-check-heatmap.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/fastqc_adapter_content_plot.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/fastqc_overrepresented_sequences_plot.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/fastqc_per_base_n_content_plot.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/fastqc_per_base_sequence_quality_plot.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Counts.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/fastqc_per_sequence_gc_content_plot_Percentages.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/fastqc_per_sequence_quality_scores_plot.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/fastqc_sequence_counts_plot-cnt.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/fastqc_sequence_counts_plot-pct.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/fastqc_sequence_duplication_levels_plot.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/fastqc_top_overrepresented_sequences_table.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/picard_deduplication-cnt.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/picard_deduplication-pct.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/samtools-flagstat-pct-table.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/samtools-flagstat-table.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/samtools-idxstats-mapped-reads-plot_Normalised_Counts-cnt.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/samtools-idxstats-mapped-reads-plot_Normalised_Counts-log.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/samtools-idxstats-mapped-reads-plot_Observed_over_Expected_Counts-cnt.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/samtools-idxstats-mapped-reads-plot_Observed_over_Expected_Counts-log.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/samtools-idxstats-mapped-reads-plot_Raw_Counts-cnt.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/samtools-idxstats-mapped-reads-plot_Raw_Counts-log.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/samtools-stats-dp.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/samtools_alignment_plot-cnt.pdf",
                "multiqc/bwamem/multiqc_plots/pdf/samtools_alignment_plot-pct.pdf",
                "multiqc/bwamem/multiqc_plots/png",
                "multiqc/bwamem/multiqc_plots/png/cutadapt_filtered_reads_plot-cnt.png",
                "multiqc/bwamem/multiqc_plots/png/cutadapt_filtered_reads_plot-pct.png",
                "multiqc/bwamem/multiqc_plots/png/cutadapt_trimmed_sequences_plot_3_Counts.png",
                "multiqc/bwamem/multiqc_plots/png/cutadapt_trimmed_sequences_plot_3_Obs_Exp.png",
                "multiqc/bwamem/multiqc_plots/png/fastqc-status-check-heatmap.png",
                "multiqc/bwamem/multiqc_plots/png/fastqc_adapter_content_plot.png",
                "multiqc/bwamem/multiqc_plots/png/fastqc_overrepresented_sequences_plot.png",
                "multiqc/bwamem/multiqc_plots/png/fastqc_per_base_n_content_plot.png",
                "multiqc/bwamem/multiqc_plots/png/fastqc_per_base_sequence_quality_plot.png",
                "multiqc/bwamem/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Counts.png",
                "multiqc/bwamem/multiqc_plots/png/fastqc_per_sequence_gc_content_plot_Percentages.png",
                "multiqc/bwamem/multiqc_plots/png/fastqc_per_sequence_quality_scores_plot.png",
                "multiqc/bwamem/multiqc_plots/png/fastqc_sequence_counts_plot-cnt.png",
                "multiqc/bwamem/multiqc_plots/png/fastqc_sequence_counts_plot-pct.png",
                "multiqc/bwamem/multiqc_plots/png/fastqc_sequence_duplication_levels_plot.png",
                "multiqc/bwamem/multiqc_plots/png/fastqc_top_overrepresented_sequences_table.png",
                "multiqc/bwamem/multiqc_plots/png/picard_deduplication-cnt.png",
                "multiqc/bwamem/multiqc_plots/png/picard_deduplication-pct.png",
                "multiqc/bwamem/multiqc_plots/png/samtools-flagstat-pct-table.png",
                "multiqc/bwamem/multiqc_plots/png/samtools-flagstat-table.png",
                "multiqc/bwamem/multiqc_plots/png/samtools-idxstats-mapped-reads-plot_Normalised_Counts-cnt.png",
                "multiqc/bwamem/multiqc_plots/png/samtools-idxstats-mapped-reads-plot_Normalised_Counts-log.png",
                "multiqc/bwamem/multiqc_plots/png/samtools-idxstats-mapped-reads-plot_Observed_over_Expected_Counts-cnt.png",
                "multiqc/bwamem/multiqc_plots/png/samtools-idxstats-mapped-reads-plot_Observed_over_Expected_Counts-log.png",
                "multiqc/bwamem/multiqc_plots/png/samtools-idxstats-mapped-reads-plot_Raw_Counts-cnt.png",
                "multiqc/bwamem/multiqc_plots/png/samtools-idxstats-mapped-reads-plot_Raw_Counts-log.png",
                "multiqc/bwamem/multiqc_plots/png/samtools-stats-dp.png",
                "multiqc/bwamem/multiqc_plots/png/samtools_alignment_plot-cnt.png",
                "multiqc/bwamem/multiqc_plots/png/samtools_alignment_plot-pct.png",
                "multiqc/bwamem/multiqc_plots/svg",
                "multiqc/bwamem/multiqc_plots/svg/cutadapt_filtered_reads_plot-cnt.svg",
                "multiqc/bwamem/multiqc_plots/svg/cutadapt_filtered_reads_plot-pct.svg",
                "multiqc/bwamem/multiqc_plots/svg/cutadapt_trimmed_sequences_plot_3_Counts.svg",
                "multiqc/bwamem/multiqc_plots/svg/cutadapt_trimmed_sequences_plot_3_Obs_Exp.svg",
                "multiqc/bwamem/multiqc_plots/svg/fastqc-status-check-heatmap.svg",
                "multiqc/bwamem/multiqc_plots/svg/fastqc_adapter_content_plot.svg",
                "multiqc/bwamem/multiqc_plots/svg/fastqc_overrepresented_sequences_plot.svg",
                "multiqc/bwamem/multiqc_plots/svg/fastqc_per_base_n_content_plot.svg",
                "multiqc/bwamem/multiqc_plots/svg/fastqc_per_base_sequence_quality_plot.svg",
                "multiqc/bwamem/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Counts.svg",
                "multiqc/bwamem/multiqc_plots/svg/fastqc_per_sequence_gc_content_plot_Percentages.svg",
                "multiqc/bwamem/multiqc_plots/svg/fastqc_per_sequence_quality_scores_plot.svg",
                "multiqc/bwamem/multiqc_plots/svg/fastqc_sequence_counts_plot-cnt.svg",
                "multiqc/bwamem/multiqc_plots/svg/fastqc_sequence_counts_plot-pct.svg",
                "multiqc/bwamem/multiqc_plots/svg/fastqc_sequence_duplication_levels_plot.svg",
                "multiqc/bwamem/multiqc_plots/svg/fastqc_top_overrepresented_sequences_table.svg",
                "multiqc/bwamem/multiqc_plots/svg/picard_deduplication-cnt.svg",
                "multiqc/bwamem/multiqc_plots/svg/picard_deduplication-pct.svg",
                "multiqc/bwamem/multiqc_plots/svg/samtools-flagstat-pct-table.svg",
                "multiqc/bwamem/multiqc_plots/svg/samtools-flagstat-table.svg",
                "multiqc/bwamem/multiqc_plots/svg/samtools-idxstats-mapped-reads-plot_Normalised_Counts-cnt.svg",
                "multiqc/bwamem/multiqc_plots/svg/samtools-idxstats-mapped-reads-plot_Normalised_Counts-log.svg",
                "multiqc/bwamem/multiqc_plots/svg/samtools-idxstats-mapped-reads-plot_Observed_over_Expected_Counts-cnt.svg",
                "multiqc/bwamem/multiqc_plots/svg/samtools-idxstats-mapped-reads-plot_Observed_over_Expected_Counts-log.svg",
                "multiqc/bwamem/multiqc_plots/svg/samtools-idxstats-mapped-reads-plot_Raw_Counts-cnt.svg",
                "multiqc/bwamem/multiqc_plots/svg/samtools-idxstats-mapped-reads-plot_Raw_Counts-log.svg",
                "multiqc/bwamem/multiqc_plots/svg/samtools-stats-dp.svg",
                "multiqc/bwamem/multiqc_plots/svg/samtools_alignment_plot-cnt.svg",
                "multiqc/bwamem/multiqc_plots/svg/samtools_alignment_plot-pct.svg",
                "multiqc/bwamem/multiqc_report.html",
                "picard",
                "picard/Ecoli_10K_methylated.bam",
                "picard/SRR389222_sub1.bam",
                "picard/SRR389222_sub2.bam",
                "picard/SRR389222_sub3.bam",
                "pipeline_info",
                "pipeline_info/nf_core_methylseq_software_mqc_versions.yml",
                "rastair",
                "rastair/call",
                "rastair/call/Ecoli_10K_methylated.rastair_call.txt",
                "rastair/call/Ecoli_10K_methylated.rastair_methylkit.txt.gz",
                "rastair/mbias",
                "rastair/mbias/Ecoli_10K_methylated.rastair_mbias.txt",
                "rastair/mbias_parser",
                "rastair/mbias_parser/Ecoli_10K_methylated.rastair_mbias_processed.csv",
                "rastair/methylkit",
                "rastair/methylkit/Ecoli_10K_methylated.rastair_methylkit.txt.gz",
                "samtools",
                "samtools/Ecoli_10K_methylated.idxstats",
                "samtools/SRR389222_sub1.idxstats",
                "samtools/SRR389222_sub2.idxstats",
                "samtools/SRR389222_sub3.idxstats",
                "trimgalore",
                "trimgalore/fastqc",
                "trimgalore/fastqc/Ecoli_10K_methylated_1_val_1_fastqc.html",
                "trimgalore/fastqc/Ecoli_10K_methylated_2_val_2_fastqc.html",
                "trimgalore/fastqc/SRR389222_sub1_trimmed_fastqc.html",
                "trimgalore/fastqc/SRR389222_sub2_trimmed_fastqc.html",
                "trimgalore/fastqc/SRR389222_sub3_trimmed_fastqc.html",
                "trimgalore/fastqc/zips",
                "trimgalore/fastqc/zips/Ecoli_10K_methylated_1_val_1_fastqc.zip",
                "trimgalore/fastqc/zips/Ecoli_10K_methylated_2_val_2_fastqc.zip",
                "trimgalore/fastqc/zips/SRR389222_sub1_trimmed_fastqc.zip",
                "trimgalore/fastqc/zips/SRR389222_sub2_trimmed_fastqc.zip",
                "trimgalore/fastqc/zips/SRR389222_sub3_trimmed_fastqc.zip",
                "trimgalore/logs",
                "trimgalore/logs/Ecoli_10K_methylated_1.fastq.gz_trimming_report.txt",
                "trimgalore/logs/Ecoli_10K_methylated_2.fastq.gz_trimming_report.txt",
                "trimgalore/logs/SRR389222_sub1.fastq.gz_trimming_report.txt",
                "trimgalore/logs/SRR389222_sub2.fastq.gz_trimming_report.txt",
                "trimgalore/logs/SRR389222_sub3.fastq.gz_trimming_report.txt"
            ],
            [
                "Ecoli_10K_methylated.bam:md5,564f154b41d9a1f88716be1a6d823fbe",
                "SRR389222_sub1.bam:md5,078262f150a6b8f6c12398e79a4a7432",
                "SRR389222_sub2.bam:md5,c91f1f8517a11641638337883e74efb7",
                "SRR389222_sub3.bam:md5,bfb0c8ff4cc448d416541d033093452d",
                "genome.amb:md5,c5a18088a0c677c9cdcebec36f2cf418",
                "genome.ann:md5,78b2fd32c283fbf8ba7624b883349a57",
                "genome.bwt:md5,86b95928f5c901663ce25d41631a97db",
                "genome.pac:md5,27c59f7f0708bb35ad99f502c2b33ab7",
                "genome.sa:md5,85d7f8991619c22bd892471e036b8069",
                "Ecoli_10K_methylated.flagstat:md5,6875b2d49eec4f8b199bceda4f3ab148",
                "Ecoli_10K_methylated.stats:md5,f41cd740027e03d823e81a509c6f3472",
                "SRR389222_sub1.flagstat:md5,fbfec3e5592cbb11f050ed06983f2bbd",
                "SRR389222_sub1.stats:md5,ddebd32ea61c93c16b19e6d685b87964",
                "SRR389222_sub2.flagstat:md5,0ad57bb4ef7b258fc23d9b3ab7ed26b4",
                "SRR389222_sub2.stats:md5,134a4bcbeb0a82d015efb83c4961ae73",
                "SRR389222_sub3.flagstat:md5,c6e3f93f66e47ff2138f640eaebd1c3b",
                "SRR389222_sub3.stats:md5,fefeb0029fe204e77b8038a2a00701b4",
                "cutadapt_filtered_reads_plot.txt:md5,ff1b6d4011a5be5410b9404f7ff56c56",
                "cutadapt_trimmed_sequences_plot_3_Counts.txt:md5,57d07c19b0c30dc88df96c30bedcdc33",
                "cutadapt_trimmed_sequences_plot_3_Obs_Exp.txt:md5,1e5ca4193f8290c5c812334d03615b41",
                "fastqc-status-check-heatmap.txt:md5,218c2a7efbb9c02f652fa78fc739c866",
                "fastqc_adapter_content_plot.txt:md5,48d8bd5ece984a74972ce0a0e814dd07",
                "fastqc_overrepresented_sequences_plot.txt:md5,9c4b7064950b5b2c486d556984f21e9e",
                "fastqc_per_base_n_content_plot.txt:md5,019d5cd4f61a31f4e7016140ca8ee36e",
                "fastqc_per_base_sequence_quality_plot.txt:md5,e4c7d522114eab9bc9101e7ee9d83c8f",
                "fastqc_per_sequence_gc_content_plot_Counts.txt:md5,302659bee8a02b6a38b24fe25f8eaa87",
                "fastqc_per_sequence_gc_content_plot_Percentages.txt:md5,c75b396eb5dda8682fb1652a9b2e92a3",
                "fastqc_per_sequence_quality_scores_plot.txt:md5,548c593f4a8be2d8627fc3bd1c280611",
                "fastqc_sequence_counts_plot.txt:md5,58eeeb06bc9dcc1764c7daf0847d6bbf",
                "fastqc_sequence_duplication_levels_plot.txt:md5,9e5712db9aa39b5754ea1e0c49cfff15",
                "fastqc_top_overrepresented_sequences_table.txt:md5,6dfc5334a09c05392ea17369aa742d01",
                "multiqc_citations.txt:md5,ad6c1e326b4276501a3b42eaad747de1",
                "multiqc_cutadapt.txt:md5,883891ac4901385102be053212068732",
                "multiqc_picard_dups.txt:md5,d55cf05658f0502ffbe45fa497627b73",
                "multiqc_samtools_flagstat.txt:md5,0d63d64b2505227942e242ea7161784f",
                "multiqc_samtools_idxstats.txt:md5,dc18fb4da0d25d75a9b4fbecc0ec023b",
                "multiqc_samtools_stats.txt:md5,14a5c75559585110bb9804edb8dcedcc",
                "picard_MarkIlluminaAdapters_histogram.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "picard_MeanQualityByCycle_histogram.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "picard_QualityScoreDistribution_histogram.txt:md5,d41d8cd98f00b204e9800998ecf8427e",
                "picard_deduplication.txt:md5,1cf872a93c4e7c5dfa9bcaba5e5c9609",
                "samtools-flagstat-pct-table.txt:md5,e791dcd246ed65ee26c77d8e51cc6eff",
                "samtools-flagstat-table.txt:md5,16819a66667e73cf53cafd3319767452",
                "samtools-idxstats-mapped-reads-plot_Normalised_Counts.txt:md5,7e6d6ed99de572d894d4fe8387882e73",
                "samtools-idxstats-mapped-reads-plot_Observed_over_Expected_Counts.txt:md5,7e6d6ed99de572d894d4fe8387882e73",
                "samtools-idxstats-mapped-reads-plot_Raw_Counts.txt:md5,474c4c51b3f330261d1b8a9c9995e4d8",
                "samtools-stats-dp.txt:md5,ee0270689f78b57905ebd812bb87891b",
                "samtools_alignment_plot.txt:md5,3a748a90ccfbb3e55811b71bcb1eece8",
                "Ecoli_10K_methylated.bam:md5,2e370f1233a364c8942c5a0250d23b3e",
                "SRR389222_sub1.bam:md5,c69a10f00a610ac1ed552a1b100f69a2",
                "SRR389222_sub2.bam:md5,456398b0cc6d09ca229eb6e5b55b04d6",
                "SRR389222_sub3.bam:md5,d58bbd13b1d016f9eb147cfa5588052d",
                "Ecoli_10K_methylated.rastair_call.txt:md5,23f789a8fe5a457e368f36e02a87ecb2",
                "Ecoli_10K_methylated.rastair_methylkit.txt.gz:md5,90fdfdb464f2a375c96bf64ea2d30445",
                "Ecoli_10K_methylated.rastair_mbias.txt:md5,63f016218f9c85a24526a9e68c97c5b7",
                "Ecoli_10K_methylated.rastair_mbias_processed.csv:md5,e46ba9ffc77f03688559a3007e005d91",
                "Ecoli_10K_methylated.rastair_methylkit.txt.gz:md5,90fdfdb464f2a375c96bf64ea2d30445",
                "Ecoli_10K_methylated.idxstats:md5,bfccbb1237079763a8b6625821423ed0",
                "SRR389222_sub1.idxstats:md5,47eaa730cb2000975476b03df209a5e8",
                "SRR389222_sub2.idxstats:md5,766288f18a1767e4b95c7383005161e8",
                "SRR389222_sub3.idxstats:md5,59fa8ca802e25072377a7527a6801f9b"
            ],
            [
                [
                    "Ecoli_10K_methylated.bam",
                    "ab11b56e62ca10400198aeaf7ff734e3"
                ],
                [
                    "SRR389222_sub1.bam",
                    "afd6b0434556e362a677a5f6bf75e214"
                ],
                [
                    "SRR389222_sub2.bam",
                    "13bf6e409b464c7ef60dc8d2757daa9c"
                ],
                [
                    "SRR389222_sub3.bam",
                    "7934578fa655119e9ab995fe8e793f44"
                ],
                [
                    "Ecoli_10K_methylated.dedup.sorted.bam",
                    "ab11b56e62ca10400198aeaf7ff734e3"
                ],
                [
                    "Ecoli_10K_methylated.deduplicated.sorted.bam",
                    "ab11b56e62ca10400198aeaf7ff734e3"
                ],
                [
                    "SRR389222_sub1.dedup.sorted.bam",
                    "f50eeee992455a3bce3c49e8485bc4ab"
                ],
                [
                    "SRR389222_sub1.deduplicated.sorted.bam",
                    "afd6b0434556e362a677a5f6bf75e214"
                ],
                [
                    "SRR389222_sub2.dedup.sorted.bam",
                    "3f75d58551db511d2e8ae7c551b3ee56"
                ],
                [
                    "SRR389222_sub2.deduplicated.sorted.bam",
                    "13bf6e409b464c7ef60dc8d2757daa9c"
                ],
                [
                    "SRR389222_sub3.dedup.sorted.bam",
                    "4afbf45a8bc5b582a3ebb698fbf3a457"
                ],
                [
                    "SRR389222_sub3.deduplicated.sorted.bam",
                    "7934578fa655119e9ab995fe8e793f44"
                ],
                [
                    "Ecoli_10K_methylated.bam",
                    "ab11b56e62ca10400198aeaf7ff734e3"
                ],
                [
                    "SRR389222_sub1.bam",
                    "afd6b0434556e362a677a5f6bf75e214"
                ],
                [
                    "SRR389222_sub2.bam",
                    "13bf6e409b464c7ef60dc8d2757daa9c"
                ],
                [
                    "SRR389222_sub3.bam",
                    "7934578fa655119e9ab995fe8e793f44"
                ]
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "24.10.5"
        },
        "timestamp": "2025-08-27T21:33:58.67893594"
    }
}